tools/building.py

```diff
@@ -1093,8 +1095,12 @@ def add_to_path(dirname):
       CLOSURE_EXTERNS += [path_from_root('src', 'minimal_runtime_worker_externs.js')]
     outfile = filename + '.cc.js'
 
+    if Settings.ALLOW_ES6:
+      lang = 'ECMASCRIPT6'
+    else:
+      lang = 'ECMASCRIPT5'
```

Comment:
this might be a separate PR. I started to experiment with it a while back,

https://github.com/emscripten-core/emscripten/compare/es6?expand=1

and ran into issues and then ran out of time.

One issue is that closure with ES6 will always emit "use strict", which we need to remove (and see that diff, it's a little tricky because of the license block we also usually remove).

I also think it can be an internal setting. We have `MIN_CHROME_*` etc. flags, and can infer from them if ES6 is allowed.